distributed/diagnostics/worker_plugin.py

```diff
@@ -0,0 +1,55 @@
+class WorkerPlugin(object):
+    """ Interface to extend the Worker
+
+    Throughout the lifecycle of a task (see :doc:`Worker <worker>`), Workers are
+    instructed by the scheduler to compute certain tasks, resulting in transitions
+    in the state of each task. The Worker owning the task is then notified of this
+    state transition.
```

Comment:
This introductory paragraph focuses on the transition method, but I suspect that many people today will also care about setup/teardown.  It might be good to have a general introductory paragraph, followed by the setup/teardown and then the transition use case.